internal/util/fetch/fetch_test.go

```diff
@@ -433,6 +426,69 @@ func TestCommand_Run_tag(t *testing.T) {
 	})
 }
 
+func TestCommand_Run_subdir_at_tag(t *testing.T) {
+	dir := "/java/expected"
+	tag := "java/v2"
+	g, rw, clean := testutil.SetupRepoAndWorkspace(t, testutil.Content{
+		Pkg: pkgbuilder.NewRootPkg().
+			WithSubPackages(pkgbuilder.NewSubPkg("java").
+				WithResource("deployment").
+				WithSubPackages(pkgbuilder.NewSubPkg("expected").WithFile("expected.txt", "My kptfile and I should be the only objects"))),
+		Branch: "main",
+		Tag:    tag,
+	})
+
+	defer clean()
+
+	err := createKptfile(rw, &kptfilev1alpha2.Git{
+		Repo:      g.RepoDirectory,
+		Directory: dir,
+		Ref:       "java/v2",
+	}, kptfilev1alpha2.ResourceMerge)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	commit, err := g.GetCommit()
+	assert.NoError(t, err)
+
+	actualPkg := pkgtesting.CreatePkgOrFail(t, rw.FullPackagePath())
+	err = Command{
+		Pkg: actualPkg,
+	}.Run(fake.CtxWithNilPrinter())
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	g.AssertEqual(t, rw.WorkspaceDirectory, actualPkg.UniquePath.String(), false)
```

Comment:
I think this could be a more compact way to check both that the content of the package is as expected and the content of `UpstreamLock` is correct:
https://github.com/GoogleContainerTools/kpt/blob/029b577ab66de50f4a988c1e2d5bee2ee68c53f6/internal/util/get/get_test.go#L1379-L1393

Reply:
Switched.